core/version.h

```diff
@@ -68,4 +68,6 @@
 // Example: "Godot v3.1.4.stable.official.mono"
 #define VERSION_FULL_NAME "" VERSION_NAME " v" VERSION_FULL_BUILD
 
+extern const char *const VERSION_HASH;
```

Comment:
Feedback welcome on whether that's the best/canonical way to declare a constant C-style string.

Reply:
I think in this specific case (splitting declaration and definition to reduce rebuilt files) this is the only/best way since `inline`/`constexpr` won't work here.

Reply:
I think this is a common approach.

* Blender: [creator_intern.h](https://github.com/blender/blender/blob/594f47ecd2d5367ca936cf6fc6ec8168c2b360d0/source/creator/creator_intern.h#L67-L82) / [buildinfo.c](https://github.com/blender/blender/blob/594f47ecd2d5367ca936cf6fc6ec8168c2b360d0/source/creator/buildinfo.c)
* Inkscape: [inkscape-version.h](https://gitlab.com/inkscape/inkscape/-/blob/a6394bdcc14be16b3d06748732f6069bba0cfabe/src/inkscape-version.h) / [inkscape-version.cpp.in](https://gitlab.com/inkscape/inkscape/-/blob/a6394bdcc14be16b3d06748732f6069bba0cfabe/src/inkscape-version.cpp.in)